app/controllers/oidc_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class OidcController < RestController
+  include FindResource
+  include AssumedRole
+  include CurrentUser
+
+
+  def index
+    # Rails 5 requires parameters to be explicitly permitted before converting
+    # to Hash.  See: https://stackoverflow.com/a/46029524
+    allowed_params = %i[account service_id]
+
+    begin
+      scope =  authenticators(
+        role: assumed_role(query_role),
+        repository: DB::Repository::AuthenticatorRepository.new,
+        handler: Authentication::Handler::OidcAuthenticationHandler.new,
+        **options(allowed_params)
+      )
+    rescue ApplicationController::Forbidden
+      raise
+    rescue ArgumentError => e
+      raise ApplicationController::UnprocessableEntity, e.message
+    end
+
+    render(json: scope)
+  end
+
+  # The v5 API currently sends +acting_as+ when listing resources
+  # for a role other than the current user.
+  def query_role
+    params[:role].presence || params[:acting_as].presence
+  end
+
+  def options(allowed_params)
+    params.permit(*allowed_params)
+          .slice(*allowed_params).to_h.symbolize_keys
+  end
+
+  def authenticators(role:, repository:, handler:, account:, service_id: nil)
```

Comment:
Let's push this logic up to a business object.  It's super hard to test here and it's causing this method to do two distinctly separate tasks (return a list of provider, returning a full provider).